app/services/prosecution_case_search.rb

```diff
@@ -34,8 +38,32 @@ def person_defendant_by_nino
     PersonDefendant.joins(:person).where(people: { nationalInsuranceNumber: permitted_params[:nationalInsuranceNumber] })
   end
 
+  def prosecution_cases_by_summons
+    ProsecutionCase.joins(:defendants).where(defendants: { defendable_type: 'PersonDefendant', defendable_id: person_defendant_by_summons })
+  end
+
+  def person_defendant_by_summons
+    PersonDefendant.where(arrestSummonsNumber: permitted_params[:arrestSummonsNumber])
+  end
+
+  def prosecution_cases_by_name_and_dob
+    ProsecutionCase.joins(:defendants).where(defendants: {
+                                               defendable_type: 'PersonDefendant',
+                                               defendable_id: person_defendant_by_name_and_dob
+                                             })
```

Comment:
Seeing this and line 42 being basically used for the same:
`ProsecutionCase.joins(:defendants).where(defendants: { defendable_type: 'PersonDefendant', defendable_id: XXX })`
... and they encose a somewhat complex query... have you considered the possibility of turning this into a `scope` in `ProsecusionCase` model (something like `ProsecusionCase.by_defendant(...)` or whatever makes sense) like has been done in `PersonDefendant` model?

Reply:
Good point, I think it might be worth improving this.